adapters/handlers/rest/handlers_authn.go

```diff
@@ -0,0 +1,68 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package rest
+
+import (
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/sirupsen/logrus"
+
+	cerrors "github.com/weaviate/weaviate/adapters/handlers/rest/errors"
+	"github.com/weaviate/weaviate/adapters/handlers/rest/operations"
+	"github.com/weaviate/weaviate/adapters/handlers/rest/operations/users"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/usecases/auth/authorization"
+	authzConv "github.com/weaviate/weaviate/usecases/auth/authorization/conv"
+)
+
+type authNHandlers struct {
+	authzController authorization.Controller
+	logger          logrus.FieldLogger
+}
+
+func setupAuthnHandlers(api *operations.WeaviateAPI, controller authorization.Controller, logger logrus.FieldLogger,
+) {
+	h := &authNHandlers{authzController: controller, logger: logger}
+
+	// user handlers
+	api.UsersGetOwnInfoHandler = users.GetOwnInfoHandlerFunc(h.getOwnInfo)
+}
+
+func (h *authNHandlers) getOwnInfo(_ users.GetOwnInfoParams, principal *models.Principal) middleware.Responder {
+	existingRoles, err := h.authzController.GetRolesForUser(principal.Username)
```

Comment:
what happens with anon access? Then the principal should be nil and we're getting a panic?